### DISCUSSION TITLE
How to make use of GoPro Labs GPS DOP levels

### DISCUSSION THREAD

**Discussion by pjmara** [gopro/labs#761] [Q&A]
Hi there,
I'm kind of cross posting between here and the traditional gopro forums, see [this post](https://community.gopro.com/s/question/0D5Uv000003QuOwKAK/query-http-api-for-if-any-gps-metadata-will-be-writing-to-recorded-videos?language=en_US)
I want to know if my camera is receiving a decent GPS signal...being able to know the DOP via the HTTP API is the goal, that way I can keep the control in python code making web requests. So somehow storing the DOP somewhere in the system status would be a great functionality, I just don't know how to do this (or if it is possible) in Labs. It would also be acceptable to flip some other setting that the HTTP API can check for, and then the HTTP API can flip the setting back and consider the camera ready to record

The bottom line is having good GPS time is what makes the GoPro "ready to record" in my use case, but I also need to verify a couple of other sensors are ready to record before actually starting a recording. So the reason I want the HTTP API to have access is so that I can start the recording when other sensors in my system are also ready- but I haven't been able to expose anything more than the 1 or 0 of GPS lock. Anyone got any tips for me?
Also, while permanent settings are viable solutions, my use case does not allow for QR codes to be scanned "at run time" 

Lastly, is there any documentation on what DOP the current system considers in Lock vs not in Lock? I cannot find any resources for this number.

**Comment by dnewman-gpsw** [comment]
Labs can make capture triggers dependent on the DOP.  However there is no way to control this via Open GoPro (HTTP API.) There is a request to have OpenGoPro support Labs commands. You can run Open GoPro with a Labs enabled cameras, so you might be able to hack together a solution, but I don't know of an easy path.

If you look at the metadata you can see when the GPS track goes from no-lock, 2D lock, to 3D Lock and read the DOP values with the GPS9 metadata stream (HERO11, or GPS5 for HERO10 and older.)  One of those will match the 1 or 0 you are seeing.

**Reply by pjmara** [reply]
This is great info- one more question that might be more answerable...
How about the limits for what GoPro labs consider gps lock "okay", "good", and "precise"? I saw this as an option in the visual builder. Are these also based on DOP?

**Reply by dnewman-gpsw** [reply]
I couldn't find or remember where gps lock "okay", "good", and "precise" is listed, but it would have been arbitrary level from DOP.
